src/proj.h

```diff
@@ -559,27 +568,490 @@ PJ_OBJ_LIST PROJ_DLL *proj_obj_create_from_name(PJ_CONTEXT *ctx,
+
+
+/** Describe how grid availability is used. */
+typedef enum {
+    /** Grid availability is only used for sorting results. Operations
+        * where some grids are missing will be sorted last. */
+    PROJ_GRID_AVAILABILITY_USED_FOR_SORTING,
+
+    /** Completely discard an operation if a required grid is missing. */
+    PROJ_GRID_AVAILABILITY_DISCARD_OPERATION_IF_MISSING_GRID,
+
+    /** Ignore grid availability at all. Results will be presented as if
+        * all grids were available. */
+    PROJ_GRID_AVAILABILITY_IGNORED,
+} PROJ_GRID_AVAILABILITY_USE;
+
+void PROJ_DLL proj_operation_factory_context_set_grid_availability_use(
+                                            PJ_OPERATION_FACTORY_CONTEXT *ctxt,
+                                            PROJ_GRID_AVAILABILITY_USE use);
+
+void PROJ_DLL proj_operation_factory_context_set_use_proj_alternative_grid_names(
+    PJ_OPERATION_FACTORY_CONTEXT *ctxt,
+    int usePROJNames);
+
+void PROJ_DLL proj_operation_factory_context_set_allow_use_intermediate_crs(
+    PJ_OPERATION_FACTORY_CONTEXT *ctxt, int allow);
+
+void PROJ_DLL proj_operation_factory_context_set_allowed_intermediate_crs(
+    PJ_OPERATION_FACTORY_CONTEXT *ctxt,
+    const char* const *list_of_auth_name_codes);
+
+/* ------------------------------------------------------------------------- */
+
+
+PJ_OBJ_LIST PROJ_DLL *proj_obj_create_operations(
+                            const PJ_OBJ *source_crs,
+                            const PJ_OBJ *target_crs,
+                            const PJ_OPERATION_FACTORY_CONTEXT *operationContext);
+
+int PROJ_DLL proj_obj_list_get_count(const PJ_OBJ_LIST *result);
+
+PJ_OBJ PROJ_DLL *proj_obj_list_get(const PJ_OBJ_LIST *result,
+                                           int index);
+
+void PROJ_DLL proj_obj_list_unref(PJ_OBJ_LIST *result);
+
+/* ------------------------------------------------------------------------- */
+
+PJ_OBJ PROJ_DLL *proj_obj_crs_get_geodetic_crs(const PJ_OBJ *crs);
+
+PJ_OBJ PROJ_DLL *proj_obj_crs_get_horizontal_datum(const PJ_OBJ *crs);
+
+PJ_OBJ PROJ_DLL *proj_obj_crs_get_sub_crs(const PJ_OBJ *crs, int index);
+
+PJ_OBJ PROJ_DLL *proj_obj_crs_get_datum(const PJ_OBJ *crs);
+
+PJ_OBJ PROJ_DLL *proj_obj_crs_get_coordinate_system(const PJ_OBJ *crs);
+
+/** Type of coordinate system. */
+typedef enum
+{
+    PJ_CS_TYPE_UNKNOWN,
+
+    PJ_CS_TYPE_CARTESIAN,
+    PJ_CS_TYPE_ELLIPSOIDAL,
+    PJ_CS_TYPE_VERTICAL,
+    PJ_CS_TYPE_SPHERICAL,
+    PJ_CS_TYPE_ORDINAL,
+    PJ_CS_TYPE_PARAMETRIC,
+    PJ_CS_TYPE_DATETIMETEMPORAL,
+    PJ_CS_TYPE_TEMPORALCOUNT,
+    PJ_CS_TYPE_TEMPORALMEASURE
+} PJ_COORDINATE_SYSTEM_TYPE;
+
+PJ_COORDINATE_SYSTEM_TYPE PROJ_DLL proj_obj_cs_get_type(const PJ_OBJ* cs);
+
+int PROJ_DLL proj_obj_cs_get_axis_count(const PJ_OBJ *cs);
+
+int PROJ_DLL proj_obj_cs_get_axis_info(const PJ_OBJ *cs, int index,
+                                       const char **pName,
```

Comment:
You work much faster than I can think, so I only just now notice that the argument names also are en camelCase, using some sort of Hungarian notation. A leftover from GDAL I am sure. Similar to the function names this is not consistent with the rest of the API, although not as important as the function names. I feel bad asking this, but is it possible to change this also?

Reply:
With the 'p', I wanted to underline the fact that it is a pointer to a const char* name, that is an output argument, to distinguish from input arguments. I feel that if we just use "const char** name" people might be confused and not realize it is an output argument.  I'm open to other suggestions.

Reply:
Another option I can see would be to prefix the argument name with out_  : out_name ?

Reply:
I see, that makes sense. At least to me it is more clear what is meant by `out_name` compared to `pName`. I hadn't made that connection until your explanation just now. Let's go with that.

Reply:
Done

Reply:
Beautiful! Thanks.